lib/kramdown_rpf/rpf.rb

```diff
@@ -110,6 +112,40 @@ def self.convert_print_only_to_html(content)
         ::Kramdown::Document.new("<div class=\"u-print-only\">\n#{content}</div>", KRAMDOWN_OPTIONS).to_html
       end
 
+      def self.convert_quiz_to_html(quiz)
+        content = YAML_FRONT_MATTER_REGEXP.match(quiz)
+        return '' if content.nil? || content.length() < 2
+
+        details = YAML.safe_load(content[1])
+        question = details['question']
+
+        choices = YAML.safe_load(content[2])
+        radios = choices.map { |choice| radio = RADIO_REGEXP.match(choice); radio ? radio[1] : '' }.reject(&:empty?)
+        radio_inputs = radios.map.with_index(1) { 
+          |text, index|  
+          <<~HEREDOC
+            <label class="c-project-quiz__label" for="choice-#{index}">#{text}</label>
+                  <input class="c-project-quiz__input" name="quiz-choice" type="radio" id="choice-#{index}" value="choice-#{index}" />
+          HEREDOC
+        }.join("      ").strip
```

Comment:
Looks better, however multiline blocks should be enclosed in `do..end`.  You can use `#compact` to weed nils out of an array.   You'd need  to change the RADIO_REGEXP to be a bit more hungry though, to avoid empty choices or maybe start the capture with a non-space.

```ruby
      RADIO_REGEXP = /\(\\)\s*(\S.*)/
```

```ruby
        radios = choices.map do |choice| 
          radio = RADIO_REGEXP.match(choice)
          radio ? radio[1] : nil # Return nil so we can use #compact below
        end

       # A one-liner could be, but maybe this is illegible.
       radios = choices.map{ |c| RADIO_REGEXP.match(c)&.[](1) }.compact

       # Add #compact here to weed out nils
       radio_inputs = radios.compact.map.with_index(1) do |text, index|
          <<~HEREDOC
            <label class="c-project-quiz__label" for="choice-#{index}">#{text}</label>
                  <input class="c-project-quiz__input" name="quiz-choice" type="radio" id="choice-#{index}" value="choice-#{index}" />
          HEREDOC
        end

       # Call .strip.join later.. maybe in the next heredoc?
      radio_inputs = radio_inputs.join('        ').strip
```

Reply:
Isn't that just the same greediness - the `\s*` will use up all the whitespace anyway so you don't need to worry about specifying a non-space character for the choice text?

also TIL `compact`